protocols/kad/src/handler.rs

```diff
@@ -780,12 +768,32 @@ where
             return Poll::Ready(event);
         }
 
-        let num_in_progress_outbound_substreams =
-            self.outbound_substreams.len() + self.num_requested_outbound_streams;
```

Comment:
Let's retain this variable please.

Reply:
Then we'll be able to only process a single outbound stream request per poll (`while` will have to change to `if`). Is that okay?

Reply:
Ah right, my bad. Thanks for pointing this out! We should process all available ones, yes.